src/tests/build.proj

```diff
@@ -467,6 +472,72 @@
     <Message Importance="High" Text="App: $(FinalPath)" />
   </Target>
 
+  <!-- This target sets custom ILC properties to ensure correct values for the build -->
+  <Target Name="_SetCustomOSSpecificProps">
+    <PropertyGroup>
+      <_targetOS>$(TargetOS)</_targetOS>
+      <UseNativeAOTRuntime>true</UseNativeAOTRuntime>
+      <NativeLib>static</NativeLib>
+      <CustomNativeMain>true</CustomNativeMain>
+      <_IsApplePlatform Condition="'$(TargetsAppleMobile)' == 'true'">true</_IsApplePlatform>
+      <_IsiOSLikePlatform Condition="'$(TargetsAppleMobile)' == 'true'">true</_IsiOSLikePlatform>
+      <DsymUtilOptions Condition="'$(_IsApplePlatform)' == 'true'">--flat</DsymUtilOptions>
+      <_SymbolPrefix Condition="'$(_IsApplePlatform)' == 'true'">_</_SymbolPrefix>
+    </PropertyGroup>
+  </Target>
+
+  <!-- This target builds a Native AOT iOS App -->
+  <Target Name="BuildNativeAOTiOSApp" DependsOnTargets="_SetCustomOSSpecificProps;SetupOSSpecificProps">
+    <PropertyGroup>
+      <CategoryPath>$([System.IO.Path]::GetDirectoryName($([System.IO.Path]::GetDirectoryName($(_CMDDIR)))))</CategoryPath>
+      <CategoryName>$([System.IO.Path]::GetFileName($(CategoryPath)))</CategoryName>
+      <TestRelativePath>$(CategoryName)/$([System.String]::Copy('$(_CMDDIR)').Replace("$(CategoryPath)/",""))</TestRelativePath>
+      <TestName>$([System.IO.Path]::GetFileName($(_CMDDIR)))</TestName>
+      <AppName>$([System.String]::Copy('$(TestRelativePath)').Replace('/','_'))</AppName>
+      <AppleBuildDir>$(IntermediateOutputPath)\iOSApps\$(AppName)</AppleBuildDir>
+      <FinalPath>$(XUnitTestBinBase)$(TestRelativePath)\$(AppName).app</FinalPath>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <NativeDependencies>$(IntermediateOutputPath)\..\$(TestRelativePath)\$(TestName)\native\$(TestName).o</NativeDependencies>
+      <DevTeamProvisioning>-</DevTeamProvisioning>
+      <AppleGenerateAppBundle>true</AppleGenerateAppBundle>
+      <GenerateXcodeProject>true</GenerateXcodeProject>
+      <AppleBundleDir>$(AppDir)</AppleBundleDir>
+      <RunAOTCompilation>false</RunAOTCompilation>
+      <MonoForceInterpreter>false</MonoForceInterpreter>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_LinkerFlagsToDrop Include="@(NativeFramework->'-framework %(Identity)')" />
+      <LinkerArg Remove="@(_LinkerFlagsToDrop)" />
+      <ExtraAppLinkerArgs Include="@(LinkerArg)" />
+      <GlobalizationkNativeLibs Include="$(MicrosoftNetCoreAppRuntimePackDir)runtimes/$(TargetOS)-$(TargetArchitecture)/native/icudt.dat" />
+    </ItemGroup>
+
+    <RemoveDir Directories="$(AppleBuildDir)" />
+    <RemoveDir Directories="$(AppDir)" />
+    <MakeDir Directories="$(AppleBuildDir)" />
+
+    <Copy
+        SourceFiles="@(GlobalizationkNativeLibs)"
+        DestinationFolder="$(AppleBuildDir)" />
+  </Target>
+
+  <!-- This target moves app bundles to the payload directory -->
```

Comment:
Can't we just have this generated in the correct spot? (The likely reason why MSBuild doesn't support moving things is because moving files around during build makes incremental compilation really hard.)

Reply:
The `AppleAppBuilder` utilizes Xcode to generate the app bundle, and by default, the output path is within the project directory at `<app-name>/<Configuration>-iphoneos/<app-name>.app`. It might be possible to specify the custom output path and it would require custom output path parameter in the `AppleAppBuilder`.

I will check if it is supported and follow-up.